src/main/java/net/sf/jabref/exporter/FileActions.java

```diff
@@ -428,10 +429,12 @@ public static SaveSession savePartOfDatabase(BibDatabaseContext bibDatabaseConte
 
                 // Check if we must write the type definition for this
                 // entry, as well. Our criterion is that all non-standard
-                // types (*not* customized standard types) must be written.
-                EntryType tp = EntryTypes.getType(be.getType(), bibDatabaseContext.getMode());
-                if (EntryTypes.getStandardType(tp.getName(), bibDatabaseContext.getMode()) == null) {
-                    types.put(tp.getName(), tp);
+                // types (*not* all customized standard types) must be written.
+                if (! EntryTypes.getStandardType(be.getType(), bibDatabaseContext.getMode()).isPresent()) {
```

Comment:
Hmm, maybe almost the same here, but I do not really get the logic. First checking StandardType and then type?